Sources/SwiftProtobufCore/JSONDecoder.swift

```diff
@@ -511,8 +517,9 @@ internal struct JSONDecoder: Decoder {
           throw JSONDecodingError.illegalNull
         }
       } else {
-        let e: E = try scanner.nextEnumValue()
-        value.append(e)
+        if let e: E = try scanner.nextEnumValue() {
+          value.append(e)
+        }
```

Comment:
This needs some tests to make sure that, e.g.,
```
{ "field": ["enum1", "enum2", "unknown", "enum3"]}
```
gets either rejected (without the option) or decoded as
```
{ "field": ["enum1", "enum2", "enum3"]}
```